explorer-nextjs/src/components/toggleButton/ToggleButton.tsx

```diff
@@ -12,17 +18,64 @@ type Options = [Option, Option];
 const ExplorerButtonGroup = ({
   size = "small",
   options,
+  onPage,
 }: {
   size?: "small" | "medium" | "large";
   options: Options;
+  onPage: string;
 }) => {
+  const [hasEpochStarted, setHasEpochStarted] = useState(false);
+  const [loading, setLoading] = useState<string | null>(null);
+
+  const { data } = useQuery({
+    enabled: true,
+    queryKey: ["currentEpoch"],
+    queryFn: fetchCurrentEpoch,
+    refetchInterval: 30000,
+    staleTime: 30000,
+    refetchOnMount: true,
+    keepPreviousData: false,
+  });
+
+  const queryClient = useQueryClient();
+  // check epoch update
+  useEffect(() => {
+    const checkEpochStatus = () => {
+      if (!data?.dateTime) return;
+      const oneHourLater = subSeconds(new Date(data.dateTime), 30).getTime();
+      const now = Date.now();
+      setHasEpochStarted(now >= oneHourLater);
+    };
+
+    checkEpochStatus();
+    const interval = setInterval(checkEpochStatus, 60000);
```

Comment:
if this block of logic is used in other parts of the app the it would be worth making a custom hook. But before that let's figure our what it's doing and why it's needed.